server/services/DashboardService/index.js

```diff
@@ -0,0 +1,132 @@
+    this.subjectKey = 'Subject ID'
+    this.updated = ''
+  }
+
+  _assessmentsFromConfig = () =>
+    _.uniqBy(this.configuration, this.assessmentKey).map(({ analysis }) => {
+      const collectionName = this.study + this.subject + analysis
+      return {
+        assessment: analysis,
+        collection: createHash(this.salt)
+          .update(collectionName)
+          .digest(this.hex),
+      }
+    })
+
+  _dataFromAssessments = async () =>
+    await Promise.all(
+      this._assessmentsFromConfig().map(async ({ collection, assessment }) => {
+        const data = await CollectionsModel.index(this.db, collection)
+        this.configDataMap.set(assessment, data)
+      })
+    )
+
+  _calculateDashboardData = () =>
+    this.configuration.forEach(({ analysis, variable, label, ...rest }) => {
+      const assessmentData = this.configDataMap.get(analysis)
+      const variableData = _.reduce(
+        assessmentData,
+        (variableDataArray, dayData) => {
+          if (Object.hasOwn(dayData, variable)) {
+            variableDataArray.push({
+              day: dayData.day,
+              [variable]: dayData[variable],
+            })
+
+            return variableDataArray
+          }
+
+          return variableDataArray
+        },
+        []
+      )
+
+      if (variableData.length) {
+        const max = _.maxBy(variableData, variable)?.[variable] || null
+        const min = _.minBy(variableData, variable)?.[variable] || null
+        const mean = _.meanBy(variableData, variable) || null
+        const stat = [{ max, min, mean }]
+
+        this.matrixData.push({
+          analysis,
+          data: variableData,
+          label,
+          variable,
+          stat,
+          ...rest,
+        })
+      } else {
+        this.matrixData.push({
+          analysis,
+          data: [],
+          label,
+          variable,
+          stat: [],
+          ...rest,
+        })
+      }
+    })
+
+  setConsentDate = async () => {
+    const metadocAttributes = {
+      study: this.study,
+      role: this.role,
+    }
+    const metadocReference = await MetadataModel.findOne(
+      this.db,
+      metadocAttributes
+    )
+
+    if (metadocReference != null) {
+      this.updated = metadocReference.updated
+
+      if (Object.hasOwn(metadocReference, this.collection)) {
+        const { collection } = metadocReference
+        const metadoc = await CollectionsModel.index(this.db, collection)
+
+        if (metadoc.length) {
+          metadoc.forEach((document) => {
+            if (
+              document[this.subjectKey] === this.subject &&
+              (document[this.consentKey] || document[this.consentDateKey])
+            ) {
+              this.consentDate =
+                document[this.consentKey] || document[this.consentDateKey]
+            }
+          })
+        }
+      }
+    }
+  }
+
+  createDashboard = async () => {
+    await this._dataFromAssessments()
+    await this.setConsentDate()
```

Comment:
should this be a private method?

Reply:
e26508b819e80caaf8bc6d068855562cd17fdc07